File: langs/python/primes/Simple.py
import time
from math import sqrt

limit = 1000000

def get_last_prime(count):
    last_prime = 2

    # Traverse each number from 1 to N with the help of for loop
    for num in range(3, count, 2):
        is_prime = True
        limit = int(sqrt(num))

        for i in range(3, limit, 2):
            if num % i == 0:
                is_prime = False
                break

        if is_prime:
            last_prime = num

    return last_prime

if __name__ == "__main__":
    start_time_ms = int(round(time.time() * 1000))

    last_prime = get_last_prime(limit)
    print(f"Last prime: {last_prime}")
    
    end_time_ms = int(round(time.time() * 1000))
    execution_time = end_time_ms - start_time_ms
    print(f"Execution time: {execution_time}ms")

File: langs/c-sharp/primes/Simple.cs
using System;

namespace Primes
{
    public sealed class Simple
    {
        private const int NUMBER = 1000000;

        public static int PrintPrimes(int count)
        {
            int lastPrime = 2;

            for (int i = 3; i <= count; i += 2)
            {
                bool isPrime = true;
                int limit = (int)Math.Sqrt(i);

                for (int j = 3; j <= limit; j += 2)
                {
                    if (i % j == 0)
                    {
                        isPrime = false;
                        break;
                    }
                }

                if (isPrime)
                {
                    lastPrime = i;
                }
            }

            return lastPrime;
        }

        public static void Main(string[] args)
        {
            DateTime start = DateTime.Now;

            long lastPrime = PrintPrimes(NUMBER);
            Console.WriteLine("Last prime: " + lastPrime);

            DateTime end = DateTime.Now;
            TimeSpan duration = end - start;

            Console.WriteLine("Execution time: " + duration.TotalMilliseconds + "ms");
        }
    }
}
